hugr-core/src/hugr/views/sibling_subgraph.rs

```diff
@@ -624,18 +646,36 @@ fn has_other_edge<H: HugrView>(hugr: &H, node: Node, dir: Direction) -> bool {
 }
 
 /// Errors that can occur while constructing a [`SimpleReplacement`].
-#[derive(Debug, Clone, PartialEq, Eq, Error)]
+#[derive(Debug, Clone, PartialEq, Error)]
 #[non_exhaustive]
 pub enum InvalidReplacement {
     /// No DataflowParent root in replacement graph.
-    #[error("No DataflowParent root in replacement graph.")]
-    InvalidDataflowGraph,
+    #[error("The root of the replacement {node} is a {}, but only OpType::DFGs are supported.", op.name())]
+    InvalidDataflowGraph {
+        /// The node ID of the root node.
+        node: Node,
+        /// The op type of the root node.
+        op: OpType,
+    },
     /// Malformed DataflowParent in replacement graph.
-    #[error("Malformed DataflowParent in replacement graph.")]
-    InvalidDataflowParent,
+    #[error("The root of the replacement {node} is malformed. It is a {} but does not contain I/O nodes.", op.name())]
+    InvalidDataflowParent {
+        /// The node ID of the root node.
+        node: Node,
+        /// The op type of the root node.
+        op: OpType,
+    },
     /// Replacement graph boundary size mismatch.
-    #[error("Replacement graph boundary size mismatch.")]
-    InvalidSignature,
+    #[error(
+        "Replacement graph boundary size mismatch. Expected {expected}, got {}.",
```

Comment:
```suggestion
        "Replacement graph boundary type mismatch. Expected {expected}, got {}.",
```

Or maybe just "graph type" without the "boundary"

Reply:
(And the comment on the previous line too)